File: Modules/Payments/Models/PaymentMethod.php
<?php

namespace Modules\Payments\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Modules\User\Models\User;

class PaymentMethod extends Model
{
    protected $fillable = [
        'user_id',
        'card_type',
        'card_number',
        'expiry',
        'is_default',
        'gateway_token',
    ];

    protected $casts = [
        'is_default' => 'boolean',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: Solid.Api/Database/Entities/PaymentMethod.cs
namespace Solid.Api.Database.Entities;

public sealed class PaymentMethod
{
    public long Id { get; set; }

    public long UserId { get; set; }

    public string CardType { get; set; } = string.Empty;

    public string CardNumber { get; set; } = string.Empty;

    public string Expiry { get; set; } = string.Empty;

    public bool IsDefault { get; set; }

    public string? GatewayToken { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public User User { get; set; } = null!;
}
